src/adapters/github.js

```diff
@@ -159,12 +159,19 @@ class GitHub extends PjaxAdapter {
       return cb();
     }
 
-    // Check if we should show in non-code pages
-    const isPR = type === 'pull';
-    const isCodePage = !type || isPR || ['tree', 'blob', 'commit'].indexOf(type) >= 0;
-    const showInNonCodePage = this.store.get(STORE.NONCODE);
-    if (!showInNonCodePage && !isCodePage) {
-      return cb();
+    if (type) {
```

Comment:
Wonder if this preserves current logic. If `type` doesn't exist, should false into the case of "Others", in which case if "Others" is checked, the tree should show, correct?

Reply:
in what i understand, the "type" variable would be null/undefined if it's on the default repo page. I want to make it always visible on this page regard of the setting, thus i add this condition

Reply:
Default page would be `code`. So maybe change the hash code to empty string so that it matches type? Basically I want code to be more explicit. Currently too implicit, will have to stop and think about the flow.

Reply:
Default page is `code`? Does this means it would be shown on the `code page` whether the `show in code` option is check, or shown when all options are unchecked? And i admit that this is too implicit

Reply:
Yes, it should show ONLY when show in code is checked. The idea is some people might not even want to see the tree in code page, only in pull request page.

Reply:
ONLY when show in code is checked? vs Default page is `code`?
I dont get it. I'm thinking of the situation when no options are checked, the octotree disappear & can't be turn back on

Reply:
That's a good point... Hmm.. On the other hand, if people don't want to see it in code page and it still shows, would be quite weird. How about we force at least 1 option is selected? Other idea is welcome.

Reply:
Can you pick another issue to work on while I think about this?

Reply:
Sure, let me know when you think through

Reply:
> How about we force at least 1 option is selected? Other idea is welcome.

That's one option, I agree. And we need a little validation hooks on checkbox

And I have another option, it can save us from writing validation but looks a bit odd

```
Show in:
( ) Every page
( ) Only at Code / Commit page
( ) Only at Pull Request page
( ) Only at Code / Commit and Pull Requests pages
```

`( )` is ratio button. For now, I think we can go with the ratio-based approach until we have more visibility options

Reply:
Great idea @duylam.

@endyquang let's do this:

```
Enable Octotree in:
 ( ) All pages
 ( ) Code/commits & pull requests
 ( ) Code/commits
 ( ) Pull requests
``` 

Rules:

* All pages is default
* Code / commit: /ovity/octotree/<BLANK or tree/xxx or blob/xxx or commit/xxx> (but not /commits)
* Pull request: /ovity/octotree/pull/xxx (but not /pulls)

Reply:
@duylam @buunguyen What would happen if none of the above options are checked?

Reply:
They're radio options not checkboxes, so one must be checked.

Reply:
Logic should be fail-safe, i.e. even if nothing in storage, set the All pages to true.

Reply:
I get it, but with these radios, it would not be as flexible as with 4 checkboxes. I have updated and made a validation hook, let me commit it first

Reply:
Why? These radios cover ALL options if you read carefully.

Reply:
I guess, in case users want `code only` or `commit only` or `commit & PR` or `code & PR`. Are there such cases?

Reply:
I see where you come from. The idea is code / commit are treated the same as it's part of Code tab. Basically simplify the life of users a bit, at the expense of a little bit less flexible. But I think it's not a bad trade off.

Reply:
Okay i believe now. Let me update everthing & push another commit

Reply:
@buunguyen Done, please check again

Reply:
Thanks, can you screenshot? Also, please help test carefully.

Reply:
![image](https://user-images.githubusercontent.com/28825116/63148832-960c6700-c02c-11e9-8336-de3e6282cc23.png)
Yep i'll be carefull